gjkr/message.go

```diff
@@ -0,0 +1,26 @@
+package gjkr
+
+import "threshold.network/roast/ephemeral"
+
+// ephemeralPublicKeyMessage is a message payload that carries the sender's
+// ephemeral public keys generated for all other group members.
+//
+// The receiver performs ECDH on a sender's ephemeral public key intended for
+// the receiver and on the receiver's private ephemeral key, creating a symmetric
+// key used for encrypting a conversation between the sender and the receiver.
+// In case of an accusation for malicious behavior, the accusing party reveals
+// its private ephemeral key so that all the other group members can resolve the
+// accusation looking at messages exchanged between accuser and accused party.
+// To validate correctness of accuser's private ephemeral key, all group members
+// must know its ephemeral public key prior to exchanging any messages. Hence,
+// this message contains all the generated public keys and it is broadcast
+// within the group.
+type ephemeralPublicKeyMessage struct {
```

Comment:
I think we need a `sessionID` as well. If we don't want to add it in this PR, let's add a TODO at least.

Reply:
I was thinking about moving session management outside of the main protocol but after reconsidering it, I think we should have it included, as originally. It could happen inactivity or disqualification could lead to slashing (out of the scope of this protocol) and we should not make assumptions about how the messages are signed. Including session ID on the protocol level would eliminate replay attacks at the core. Added a point to the TODO.

Reply:
Added session ID filtering in 02c828c3c1e7507bfa37f7690ea2a3004c565a3d.